src/noggit/scripting/script_loader-noggit_module.ipp

```diff
@@ -0,0 +1,301 @@
+// This file is part of Noggit3, licensed under GNU General Public License (version 3).
+
+// Note: This file should only be included by script_loader.cpp
+#pragma once
+
+#include <noggit/scripting/scripting_tool.hpp>
+#include <noggit/scripting/script_image.hpp>
+#include <noggit/scripting/script_context.hpp>
+#include <noggit/scripting/script_filesystem.hpp>
+#include <noggit/scripting/script_math.hpp>
+#include <noggit/scripting/script_noise.hpp>
+#include <noggit/scripting/script_random.hpp>
+#include <noggit/scripting/script_selection.hpp>
+#include <noggit/scripting/scripting_tool.hpp>
+#include <noggit/scripting/script_brush.hpp>
+#include <noggit/scripting/script_misc.hpp>
+#include <noggit/scripting/script_chunk.hpp>
+#include <noggit/scripting/script_model.hpp>
+#include <noggit/scripting/script_vert.hpp>
+#include <math/vector_3d.hpp>
+
+#include <daScript/daScript.h>
+
+#include <string>
+
+using namespace das;
+using namespace noggit::scripting;
+using namespace math;
+
+// TODO: Rename this one, it's the odd one
+#define FUNC(name, side_effect) \
+  addExtern<DAS_BIND_FUN(name)>(*this, lib, #name, SideEffects::side_effect, #name)
+
+#define FUNC_RETVALUE(name, side_effect) \
+  addExtern<DAS_BIND_FUN(noggit::scripting::name), SimNode_ExtFuncCallAndCopyOrMove>(*this, lib, #name, SideEffects::side_effect, #name)
+
+#define FUNC_SCOPED(name, side_effect) \
+  addExtern<DAS_BIND_FUN(noggit::scripting::name)>(*this, lib, #name, SideEffects::side_effect, #name)
+
+#define FIELD(name) \
+  addField<DAS_BIND_MANAGED_FIELD(name)>(#name);
+
+#define CLASS(name, ...)                               \
+  MAKE_TYPE_FACTORY(name, name);                           \
+  struct name##_annotation : public ManagedStructureAnnotation<name, true, true>   \
+  {                                        \
+    name##_annotation(ModuleLibrary& ml) : ManagedStructureAnnotation(#name, ml) \
+    {                                      \
+      __VA_ARGS__                                \
+    }                                      \
+    virtual bool isLocal() const override { return true; }             \
+    virtual bool canCopy() const override { return true; }             \
+    virtual bool canMove() const override { return true; }             \
```

Comment:
All of the classes exposed to daScript are marked Copy and Move-able, while their c++ counterparts are not.
Can you elaborate on the rational here?

Reply:
copypaste from the module tutorial, no benefit to keeping them movable. `canCopy` must be enabled to allow creating variables at all: `var b = create_image()` involves one copy, and daScript does call the copy constructor here. I kept constructors empty from when I was unsure if C++-side refcounting would somehow be messed up in that process, but it doesn't seem like this will matter anymore.

Reply:
Actually, I realized the semantics I had planned are a little messy, but I will come back to this when I figure out the gc